src/vs_seeker.c

```diff
@@ -0,0 +1,776 @@
+        case OBJ_EVENT_GFX_BLACK_BELT:
+        case OBJ_EVENT_GFX_BOY_1:
+        case OBJ_EVENT_GFX_BOY_2:
+        case OBJ_EVENT_GFX_BOY_3:
+        case OBJ_EVENT_GFX_BUG_CATCHER:
+        case OBJ_EVENT_GFX_CAMPER:
+        case OBJ_EVENT_GFX_CYCLING_TRIATHLETE_F:
+        case OBJ_EVENT_GFX_CYCLING_TRIATHLETE_M:
+        case OBJ_EVENT_GFX_EXPERT_F:
+        case OBJ_EVENT_GFX_EXPERT_M:
+        case OBJ_EVENT_GFX_FAT_MAN:
+        case OBJ_EVENT_GFX_FISHERMAN:
+        case OBJ_EVENT_GFX_GENTLEMAN:
+        case OBJ_EVENT_GFX_GIRL_1:
+        case OBJ_EVENT_GFX_GIRL_2:
+        case OBJ_EVENT_GFX_GIRL_3:
+        case OBJ_EVENT_GFX_HEX_MANIAC:
+        case OBJ_EVENT_GFX_HIKER:
+        case OBJ_EVENT_GFX_LASS:
+        case OBJ_EVENT_GFX_LITTLE_BOY:
+        case OBJ_EVENT_GFX_LITTLE_GIRL:
+        case OBJ_EVENT_GFX_MAGMA_MEMBER_F:
+        case OBJ_EVENT_GFX_MAGMA_MEMBER_M:
+        case OBJ_EVENT_GFX_MAN_3:
+        case OBJ_EVENT_GFX_MAN_4:
+        case OBJ_EVENT_GFX_MAN_5:
+        case OBJ_EVENT_GFX_MANIAC:
+        case OBJ_EVENT_GFX_NINJA_BOY:
+        case OBJ_EVENT_GFX_PICNICKER:
+        case OBJ_EVENT_GFX_POKEFAN_F:
+        case OBJ_EVENT_GFX_POKEFAN_M:
+        case OBJ_EVENT_GFX_PSYCHIC_M:
+        case OBJ_EVENT_GFX_RICH_BOY:
+        case OBJ_EVENT_GFX_RUNNING_TRIATHLETE_F:
+        case OBJ_EVENT_GFX_RUNNING_TRIATHLETE_M:
+        case OBJ_EVENT_GFX_SAILOR:
+        case OBJ_EVENT_GFX_SCHOOL_KID_M:
+        case OBJ_EVENT_GFX_TUBER_F:
+        case OBJ_EVENT_GFX_TUBER_M:
+        case OBJ_EVENT_GFX_TWIN:
+        case OBJ_EVENT_GFX_WOMAN_1:
+        case OBJ_EVENT_GFX_WOMAN_2:
+        case OBJ_EVENT_GFX_WOMAN_4:
+        case OBJ_EVENT_GFX_WOMAN_5:
+        case OBJ_EVENT_GFX_YOUNGSTER:
+            return MOVEMENT_TYPE_ROTATE_CLOCKWISE;
+            //return MOVEMENT_TYPE_RAISE_HAND_AND_JUMP;
+        case OBJ_EVENT_GFX_SWIMMER_F:
+        case OBJ_EVENT_GFX_SWIMMER_M:
+        case OBJ_EVENT_GFX_TUBER_M_SWIMMING:
+            //return MOVEMENT_TYPE_RAISE_HAND_AND_SWIM;
+            return MOVEMENT_TYPE_ROTATE_CLOCKWISE;
+        default:
+            //return MOVEMENT_TYPE_RAISE_HAND_AND_STOP;
+            return MOVEMENT_TYPE_FACE_DOWN;
+    }
+    // BRANCH_NOTE: These lines are in Jaizu's original implementation, but have been commented out as this branch uses the behavior from Pokemon DPPt, where Trainers will spin clockwise when they can be rebattled.
+}
+
+static u16 GetTrainerFlagFromScript(const u8 *script)
+/*
+ * The trainer flag is a little-endian short located +2 from
+ * the script pointer, assuming the trainerbattle command is
+ * first in the script.  Because scripts are unaligned, and
+ * because the ARM processor requires shorts to be 16-bit
+ * aligned, this function needs to perform explicit bitwise
+ * operations to get the correct flag.
+ *
+ * 5c XX YY ZZ ...
+ *       -- --
+ */
+{
+    u16 trainerFlag;
+
+    script += 2;
+    trainerFlag = script[0];
+    trainerFlag |= script[1] << 8;
```

Comment:
`ScriptReadHalfword` and we don't need the explanation anymore, in fact this can probably be a single function call either at the caller side or inlined.

Reply:
Or rather `T1_READ_16` because this is not a Script Context :)